corehq/apps/data_interfaces/views.py

```diff
@@ -651,88 +676,76 @@ def _format_rule(self, rule):
                          .done()
                          .strftime(SERVER_DATETIME_FORMAT)) if rule.last_run else '-',
             'edit_url': reverse(EditCaseRuleView.urlname, args=[self.domain, rule.pk]),
+            'action_error': "",     # must be provided because knockout template looks for it
         }
 
-    @allow_remote_invocation
-    def get_pagination_data(self, in_data):
-        try:
-            limit = int(in_data['limit'])
-            page = int(in_data['page'])
-        except (TypeError, KeyError, ValueError):
-            return {
-                'success': False,
-                'error': _("Please provide pagination info."),
+    @property
+    def paginated_list(self):
+        for rule in self._rules()[self.skip:self.skip + self.limit]:
+            yield {
+                'itemData': self._format_rule(rule),
+                'template': 'base-rule-template',
             }
 
-        start = (page - 1) * limit
-        stop = limit * page
-
-        rules = AutomaticUpdateRule.by_domain(
+    @memoized
+    def _rules(self):
+        return AutomaticUpdateRule.by_domain(
             self.domain,
             AutomaticUpdateRule.WORKFLOW_CASE_UPDATE,
             active_only=False,
         )
```

Comment:
i always like to make sure to include an order_by in the result sets of paginating views to ensure that the pages are always determined the same way.

it looks like the old view was only ordering by name, but we should also order by id to break any ties if there are two rules with the same name, so ordering by name and id should be good.